src/decoder.rs

```diff
@@ -114,15 +110,11 @@ impl<'a> Decoder<'a> {
 
     /// read a slice of bytes determined at runtime
     pub fn read_slice(&mut self, len: usize) -> DecodeResult<&'a [u8]> {
-        let end = self
-            .index
-            .checked_add(len)
-            .ok_or(DecodeError::AddOverflow)?;
-        let slice = self
-            .buffer
-            .get(self.index..end)
-            .ok_or(DecodeError::EndOfBuffer { index: end })?;
-        self.index = end;
+        if len > self.buffer.len() {
```

Comment:
> The only potential problem is that DecodeError::EndOfBuffer is no longer constructed by this crate, because this PR removes the information required to construct it.

Just confirming, there is still no chance of panic because the condition is checked here though, right? I think `EndOfBuffer` is now no longer used anywhere, there are probably a few other variants in `DecodeError` that are no longer used we can remove... can save this for another PR though.

Reply:
Yup, same as above.